### PR TITLE
📊 Adding the WHO Mortality DB

### PR DESCRIPTION
**Pull request by spoonerf**

### CODE REVIEW

File: etl/steps/data/garden/who/2023-08-01/mortality_database.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        "[0]": "less than 1 year",
+        "[All]": "all ages",
+    }
+
+    tb["age_group"] = tb["age_group"].replace(age_dict, regex=False)
+
+    return tb
+
+
+def add_metadata(dest_dir: str, ds_meadow: Dataset, tb: Table) -> Dataset:
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/who/2023-08-01/mortality_database.py:89]: this is cool! the only thing is that it wasn't immediately clear to me that you were creating different tables here for each group. Maybe add this note to docstring? e.g.,
`  """ Returns:
   Dataset: containing tables with causes of mortality for each group ("cause", "sex", "age_group")
    """`

File: etl/steps/data/garden/who/2023-08-01/mortality_database.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            # dropping columns that are totally empty - not all combinations of variables exist or have been downloaded
+        group = group.dropna(axis=1, how="all")
+        # Dropping dims as table name contains them
+        group = group.drop(columns=dims)
+        group = group.set_index(["country", "year"], verify_integrity=True)
+        ds_garden.add(group)
+    return ds_garden
+
+
+def create_variable_metadata(
```
Comment (veronikasamborska1994) [inline etl/steps/data/garden/who/2023-08-01/mortality_database.py:123]: this is super optional and would only be mainly useful if others want to adapt your code but you could add a bit more documentation on how you create variable metadata here (ie. a slightly more extended description on what the function does). but to be honest it's also quite clear once you read it :) so completely up to you!

**Review by veronikasamborska1994 — COMMENTED** [review]
looks good! very minor comments

**Review by veronikasamborska1994 — APPROVED** [review]

File: snapshots/who/2023-08-01/mortality_database.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    for cause in list_of_causes.keys():
+        file_name = [s for s in csv_files if cause in s]
+        file_name = "".join(file_name)
+        file_path = base_file_path + file_name
+        assert file_path.endswith("csv")
+        df = pd.read_csv(file_path, skiprows=6)
+        cols = df.columns
+        df = df.reset_index()
+        df = df.dropna(how="all", axis=1)
+        df.columns = cols
```
[the remark this answers is not in the document]
Comment (spoonerf) [reply snapshots/who/2023-08-01/mortality_database.py:114]: Ah it is a bit weird, the csv was being read with the columns shifted by one to the right. Not sure why that was happening as it looked normal in Excel. It's a bit hacky and horrible but couldn't think of a better way to handle it.

File: etl/steps/data/garden/who/2023-08-01/mortality_database.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    # Read table from meadow dataset.
+    tb = ds_meadow["mortality_database"]
+    tb = tb.reset_index()
+    #
+    # Process data.
+    #
+    tb = tidy_sex_dimension(tb)
+    tb = tidy_age_dimension(tb)
+    tb = tidy_causes_dimension(tb)
+    tb: Table = geo.harmonize_countries(
```
[the remark this answers is not in the document]
Comment (spoonerf) [reply etl/steps/data/garden/who/2023-08-01/mortality_database.py:31]: Good catch, thanks!

File: etl/steps/data/garden/who/2023-08-01/mortality_database.py (hunk 2 above)
Comment (spoonerf) [reply etl/steps/data/garden/who/2023-08-01/mortality_database.py:123]: Very good point, the docstring is not very informative at the moment!

**Pull request merged**